admin/tabs/userData/roles/RoleModel.ts

```diff
@@ -79,8 +83,9 @@ export class RoleModel extends HoistModel {
 
     @action
     setRoles(roles: HoistRole[]) {
+        const gridData = this.groupingDisabled ? roles : this.processRolesForTreeGrid(roles);
         this.allRoles = roles;
-        this.gridModel.loadData(roles);
+        this.gridModel.loadData(gridData);
```

Comment:
I would  add an additional private method of the form l`displayRoles()`, that does the pre-processing and then loads the data in the grid.    `allRoles` should just be set once on this public setter.